lib/edoc/src/edoc_layout_chunks.erl

```diff
@@ -0,0 +1,512 @@
+	{_Tag, _Attrs, SubNodes} -> [chunk_to_text(SubNodes) | chunk_to_text(Nodes)]
+    end.
+
+xpath_to_atom(XPath, Doc, Opts) ->
+    binary_to_atom(xpath_to_text(XPath, Doc, Opts), utf8).
+
+xpath_to_integer(XPath, Doc, Opts) ->
+    binary_to_integer(xpath_to_text(XPath, Doc, Opts)).
+
+xpath_to_chunk(XPath, Doc) ->
+    XmerlDoc = xmerl_xpath:string(XPath, Doc),
+    xmerl_to_chunk(XmerlDoc).
+
+%%.
+%%' Xmerl to chunk format
+%%
+
+-spec xmerl_to_chunk([xmerl_doc_node()]) -> shell_docs:chunk_elements().
+xmerl_to_chunk(Contents) ->
+    shell_docs:normalize(format_content(Contents)).
+
+-spec format_content([xmerl_doc_node()]) -> shell_docs:chunk_elements().
+format_content(Contents) ->
+    {SeeTags, OtherTags} = lists:partition(fun (#xmlElement{name = see}) -> true;
+					       (_) -> false end,
+					   Contents),
+    lists:flatten([ format_content_(T) || T <- OtherTags ] ++ rewrite_see_tags(SeeTags)).
+
+-spec format_content_(xmerl_doc_node()) -> shell_docs:chunk_elements().
+format_content_(#xmlPI{})      -> [];
+format_content_(#xmlComment{}) -> [];
+
+format_content_(#xmlText{} = T) ->
+    Text = T#xmlText.value,
+    case edoc_lib:is_space(Text) of
+	true -> [];
+	false -> [unicode:characters_to_binary(Text)]
+    end;
+
+format_content_(#xmlElement{name = equiv} = E) ->
+    format_element(rewrite_equiv_tag(E));
+format_content_(#xmlElement{name = a} = E) ->
+    format_element(rewrite_a_tag(E));
+format_content_(#xmlElement{} = E) ->
+    format_element(E).
+
+format_element(#xmlElement{} = E) ->
+    #xmlElement{name = Name, content = Content, attributes = Attributes} = E,
+    case {is_edoc_tag(Name), is_html_tag(Name)} of
+	{true, _} ->
+	    format_content(Content);
+	{_, false} ->
+	    edoc_report:warning("'~s' is not accepted - skipping tag, extracting content", [Name]),
+	    format_content(Content);
+	_ ->
+	    [{Name, format_attributes(Attributes), format_content(Content)}]
+    end.
+
+-spec format_attributes([xmerl_attribute()]) -> [shell_docs:chunk_element_attr()].
+format_attributes(Attrs) ->
+    [ format_attribute(Attr) || Attr <- Attrs ].
+
+-spec format_attribute(xmerl_attribute()) -> shell_docs:chunk_element_attr().
+format_attribute(#xmlAttribute{} = Attr) ->
+    #xmlAttribute{name = Name, value = V} = Attr,
+    %% From xmerl.hrl: #xmlAttribute.value :: IOlist() | atom() | integer()
+    case V of
+	_ when is_list(V)    -> {Name, unicode:characters_to_binary(V)};
+	_ when is_atom(V)    -> {Name, atom_to_binary(V, utf8)};
+	_ when is_integer(V) -> {Name, integer_to_binary(V)}
+    end.
+
+-spec is_edoc_tag(atom()) -> boolean().
+is_edoc_tag(fullDescription) -> true;
+is_edoc_tag(since) -> true;
+is_edoc_tag(_) -> false.
+
+-spec is_html_tag(atom()) -> boolean().
+is_html_tag(Tag) ->
+    %% This is only a subset of existing HTML tags.
+    %% Compare with https://developer.mozilla.org/en-US/docs/Web/HTML/Element
+    Tags = [a,p,h1,h2,h3,i,br,em,pre,code,ul,ol,li,dl,dt,dd],
```

Comment:
Hmm, from http://erlang.org/doc/apps/edoc/chapter.html#headings and your link it seems that heading ranks up to and including `h5` might be generated by EDoc, unless somebody explicitly uses `<h6>` tag. I think it would be cleaner to let them all pass through to maintain the document structure / nesting, but maybe just do not distinguish between the 4/5/6 levels in the shell renderer?

Reply:
yes, that makes sense. Can you add it to shell_docs, or should I?

Reply:
I'll take care of it

Reply:
done in #2991